week1-syntax.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"bufio"
+	"os"
+	"errors"
+	"strings"
+	"strconv"
+)
+
+func eval(a float64, b float64, arithmetic string) (float64, error) {
+	var result float64 = 0
+	var err error = nil
+	switch arithmetic {
+	case "+":
+		result = a + b
+	case "-":
+		result = a - b
+	case "*":
+		result = a * b
+	case "/":
+		result = a / b
+	default:
+		err = errors.New("Invalid arithmetic")
+	}
+
+	return result, err
+}
+
+func main() {
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Printf("> ")
+
+	scanner.Scan()
+	text := scanner.Text()
+	s := strings.Split(text, " ")
+	if len(s) != 3 {
+		fmt.Printf("ERROR\n")
+		return
+	}
+
+	a, err1 := strconv.ParseFloat(s[0], 64)
+	if err1 != nil {
+		fmt.Printf("ERROR\n")
```

Comment:
A specific error would be better for users and yourself when you need to troubleshoot if there are any errors.